cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx

```diff
@@ -58,28 +60,44 @@ const MultiStageSelectorBase: React.FC<IMultiStageSelectorProps> = ({
 }) => {
   const inputSchema = objectQuery(extraConfig, 'inputSchema');
   const inputStages = getInputStages(inputSchema);
-  const { delimiter } = widgetProps;
-  const multiSelectWidgetProps = {
-    delimiter,
-    options: inputStages,
-    showSelectionCount: false,
-    emptyPlaceholder: 'Select input stage',
-  };
+  const { delimiter, singleSelectOnly } = widgetProps;
+
   if (!inputStages.length) {
     return (
       <div className={classes.emptyMessageContainer}>
         <div className={classes.emptyMessage}>No input stages</div>
       </div>
     );
   }
-  return (
-    <MultiSelect
-      value={value}
-      onChange={onChange}
-      widgetProps={multiSelectWidgetProps}
-      disabled={disabled}
-    />
-  );
+
+  if (singleSelectOnly || false) {
```

Comment:
I don't think we need the "or" logic here.

Reply:
right...thanks!